### ISSUE TITLE
Add option for groups to `sort-keys`

### ISSUE THREAD

**Issue by timolins** [eslint/eslint#12275]
**What rule do you want to change?**

[`sort-keys`](https://eslint.org/docs/rules/sort-keys)

**Does this change cause the rule to produce more or fewer warnings?**

Fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**

New option to allow groups separated by blank lines. This concept is taken from the [TSLint counterpart](https://palantir.github.io/tslint/rules/object-literal-sort-keys/), where this is the default behaviour. This will be very helpful for everyone who is migrating from TSLint and used this rule.

**Please provide some example code that this change will affect:**

```js
const foo = {
	a: "Value",
	b: "Bar",

	action: () => {
		console.log()
	}
}

```

**What does the rule currently do for this code?**

`action` should be before `b`.

**What will the rule do after it's changed?**

Reset the sorting after each blank line, which makes the code valid.

**Are you willing to submit a pull request to implement this change?**

I might look into it, but I can't promise anything since I do not have prior experience with writing lint rules.

**Comment by g-plane** [comment]
Thanks for this issue.

Personally, I think this feature can be implemented in `@typescript-eslint/plugin` instead of ESLint core.

**Comment by kaicataldo** [comment]
For my own understanding - the criteria for sorting isn’t being changed in this proposal, correct? It just allows you to sort individual groups (separated by a padded newline) with whatever the current sorting behavior is?

**Comment by timolins** [comment]
Yes, you are right.

**Comment by kaicataldo** [comment]
Thanks for clarifying. I’m 👍 for adding this to the core rule. I don’t think it should have to be implemented by `@typescript-eslint/plugin` since the changes proposed aren’t related to TyoeScript syntax.
